test/extension/handlers/topologymutation/handler.go

```diff
@@ -19,63 +19,225 @@ package topologymutation
+				log.Error(err, "error patching DockerClusterTemplate")
+				return errors.Wrap(err, "error patching DockerClusterTemplate")
+			}
+		}
+		if controlPlaneTemplate, ok := obj.(*controlplanev1.KubeadmControlPlaneTemplate); ok {
+			if err := patchKubeadmControlPlaneTemplate(ctx, controlPlaneTemplate, variables); err != nil {
+				log.Error(err, "error patching KubeadmControlPlaneTemplateß")
+				return errors.Wrapf(err, "error patching KubeadmControlPlaneTemplate")
+			}
+		}
+		if machineDeploymentBootstrapTemplate, ok := obj.(*bootstrapv1.KubeadmConfigTemplate); ok {
+			if err := patchMachineDeploymentBootstrapTemplate(ctx, machineDeploymentBootstrapTemplate, variables); err != nil {
+				log.Error(err, "error patching KubeadmConfigTemplate")
+				return errors.Wrap(err, "error patching KubeadmConfigTemplate")
+			}
+		}
+		if dockerMachineTemplate, ok := obj.(*infrav1.DockerMachineTemplate); ok {
+			if err := patchDockerMachineTemplate(ctx, dockerMachineTemplate, variables); err != nil {
+				log.Error(err, "error patching DockerMachineTemplate")
+				return errors.Wrap(err, "error patching DockerMachineTemplate")
 			}
 		}
-
 		return nil
 	})
 }
 
-// patchDockerClusterTemplate patches the DockerClusterTepmlate.
-func patchDockerClusterTemplate(dockerClusterTemplate *infrav1.DockerClusterTemplate, templateVariables map[string]apiextensionsv1.JSON) error {
-	// Get the variable value as JSON string.
-	value, err := patchvariables.GetVariableValue(templateVariables, "lbImageRepository")
+func patchKubeadmControlPlaneTemplate(ctx context.Context, kubeadmControlPlaneTemplate *controlplanev1.KubeadmControlPlaneTemplate, templateVariables map[string]apiextensionsv1.JSON) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	// 1) Patch .Spec.KubeadmConfigSpec.ClusterConfiguration.ImageRepository with the value from the Cluster Topology variables.
+	//    If it's not set in the variables default to defaultImageRepository.
+	// TODO: no-op if the variable doesn't exist.
+	// TODO: error aggregation.
+	if found, imageRepository, err := getStringValueForVariable(templateVariables, "imageRepository"); found {
+		if err != nil {
+			return errors.Wrap(err, "could not set spec.template.spec.kubeadmConfigSpec.clusterConfiguration.imageRepository")
+		}
+		log.Info("Setting KubeadmControlPlaneTemplate imageRepository", "imageRepository", imageRepository)
+		kubeadmControlPlaneTemplate.Spec.Template.Spec.KubeadmConfigSpec.ClusterConfiguration.ImageRepository = imageRepository
+	}
+
+	// 2) If the Kubernetes version from builtin.controlPlane.version is below 1.24.0 set "cgroup-driver": "cgroupfs" to
+	//    - kubeadmConfigSpec.InitConfiguration.KubeletExtraArgs
+	//    - kubeadmConfigSpec.JoinConfiguration.KubeletExtraArgs
+	maximumVersionForPatch, err := version.ParseMajorMinorPatchTolerant("v1.24.0")
+	if err != nil {
+		return errors.Wrap(err, "could not set parse kubernetes version v1.24.0")
```

Comment:
```suggestion
		return errors.Wrap(err, "could not parse Kubernetes version v1.24.0")
```

A must func would be really nice in the fullness of time for cases like this..

Reply:
Ah. Can we just inline the version in l.119 instead of parsing it? `semver.Version{Major: Minor: Patch}` 

(and possible move it to a file-level constant? we usually did that in CABPK/KCP)